logprep/ng/processor/list_comparison/processor.py

```diff
@@ -81,14 +81,20 @@ def _list_comparison(self, rule: ListComparisonRule, event: dict) -> tuple[list[
         the result (key) and a list of filenames pertaining to said result (value).
         """
 
+        # get value that should be checked in the lists
         field_value = get_dotted_field_value(event, rule.source_fields[0])
-
-        list_matches = [
-            compare_list
-            for compare_list in rule.compare_sets
-            if field_value in rule.compare_sets[compare_list]
-        ]
-
-        if not list_matches:
+        value_list = field_value if isinstance(field_value, list) else [field_value]
+
+        # iterate over lists and check if element is in any
```

Comment:
Same here with the comment. Rather extract a method with the name `check_for_value_in_lists` (or sth.)

Reply:
I cleaned up the comments. I previously kept them as they are without noticing that they were already removed from the ng version. I agree that it's better to not have them.